cloudinit/config/cc_power_state_change.py

```diff
@@ -161,20 +178,29 @@ def load_power_state(cfg):
             (','.join(opt_map.keys()), mode))
 
     delay = pstate.get("delay", "now")
-    # convert integer 30 or string '30' to '+30'
-    try:
-        delay = "+%s" % int(delay)
-    except ValueError:
-        pass
-
-    if delay != "now" and not re.match(r"\+[0-9]+", delay):
+    if delay != "now" and not re.match(r"^\+[0-9]+$", delay):
```

Comment:
Yes I changed this on purpose. As per the docs the only valid values for delay are either 'now' or '+MINS'. To avoid the ability to specify HH:MM (which the shutdown command itself accepts but, for example, Alpine's halt/poweroff/reboot don't accept) I tightened up the regex. I also tweaked the docs in cc_ntp.py in the one place where it mentioned that delay could use other formats handled by shutdown as this disagreed with the rest of the docs.

I could add leading "zero or more whitespace" to the start and end of the regex if you think that's a good idea.

Reply:
> I could add leading "zero or more whitespace" to the start and end of the regex if you think that's a good idea.

Yeah, or drop the change; either is fine.  I'm mostly worried about runtime behavior that could break (even if the docs say otherwise; easier to update the docs than it is to unbreak released code).

Reply:
I reworked the code in line with what we discussed on IRC.